libcrypto.mk: modifications to make it build in yocto environment

Adding an include file that is setting the common flags.

Modified the build rule so that it outputs the binary and include
to the OUT_DIR


Signed-off-by: Nisha Parrakat <Nisha.Parrakat@kpit.com>

--- git/external/boringssl/debian/libcrypto.mk	2021-04-30 11:32:40.610312307 +0000
+++ git/external/boringssl/debian/libcrypto.mk	2021-04-30 11:36:47.738318880 +0000
@@ -1,3 +1,4 @@
+include ../../rules_yocto.mk
 include sources.mk
 
 NAME = libcrypto
@@ -26,6 +27,8 @@
 LDFLAGS += -shared -Wl,-soname,$(NAME).so.0 -lpthread
 
 build: $(SOURCES)
-	mkdir --parents debian/out
-	$(CC) $^ -o debian/out/$(NAME).so.0 $(CFLAGS) $(CPPFLAGS) $(LDFLAGS)
-	ln -s $(NAME).so.0 debian/out/$(NAME).so
\ No newline at end of file
+	mkdir --parents $(OUT_DIR)/usr/lib/$(DEB_HOST_MULTIARCH)/android/ 
+	mkdir --parents $(OUT_DIR)/usr/include
+	$(CC) $^ -o $(OUT_DIR)/usr/lib/$(DEB_HOST_MULTIARCH)/android/$(NAME).so.0 $(CFLAGS) $(CPPFLAGS) $(LDFLAGS) -DOPENSSL_NO_ASM
+	ln -sf $(NAME).so.0 $(OUT_DIR)/usr/lib/$(DEB_HOST_MULTIARCH)/android/$(NAME).so
+	cp -r include/openssl $(OUT_DIR)/usr/include
